runtime/src/utils/buffer/tip.rs

```diff
@@ -128,7 +129,7 @@ impl Buffer {
     /// may result are filled with zeros. Returns `true` if the merge was performed, otherwise the
     /// caller is responsible for continuing to manage the data.
     pub(super) fn merge(&mut self, data: &[u8], offset: u64) -> bool {
-        let end_offset = offset + data.len() as u64;
+        let end_offset = offset.checked_add(data.len() as u64).expect("overflow");
```

Comment:
nit: for both of these cases, can we add more detail into "what overflowed"?

I'd prefer something like `"offset overflowed"`?